File: server/datasets/python_source-code/Multiple-E1/type2_renamed_1_grid_dfs.py
def calculate_islands(ocean_map):
    if not ocean_map:
        return 0

    def explore_land(y, x):
        if y < 0 or x < 0 or y >= len(ocean_map) or x >= len(ocean_map[0]) or ocean_map[y][x] == '0':
            return
        ocean_map[y][x] = '0'
        moves = [(1, 0), (-1, 0), (0, 1), (0, -1)]
        for dy, dx in moves:
            explore_land(y + dy, x + dx)

    total_islands = 0
    for y in range(len(ocean_map)):
        for x in range(len(ocean_map[0])):
            if ocean_map[y][x] == '1':
                total_islands += 1
                explore_land(y, x)
    return total_islands

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_e1/type2_renamed_1_grid_dfs.java
public class Solution {
    public int calculate_islands(char[][] ocean_map) {
        if (ocean_map == null || ocean_map.length == 0) {
            return 0;
        }

        int total_islands = 0;
        for (int y = 0; y < ocean_map.length; y++) {
            for (int x = 0; x < ocean_map[0].length; x++) {
                if (ocean_map[y][x] == '1') {
                    total_islands++;
                    explore_land(ocean_map, y, x);
                }
            }
        }
        return total_islands;
    }

    private void explore_land(char[][] ocean_map, int y, int x) {
        if (y < 0 || x < 0 || y >= ocean_map.length || x >= ocean_map[0].length || ocean_map[y][x] == '0') {
            return;
        }
        ocean_map[y][x] = '0';
        int[][] moves = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
        for (int[] mv : moves) {
            explore_land(ocean_map, y + mv[0], x + mv[1]);
        }
    }
}
